assets/react/main.js

```diff
@@ -3,55 +3,56 @@ var Router = require('react-router');
 var Route = Router.Route;
 var Wrapper = require('./wrapper.js');
 var Home = require('./components/home.js');
-var MyPosts = require('./components/my_posts.js');
+var Profile = require('./components/profile.js');
 var EditPost = require('./components/edit_post.js');
 var ShowPost = require('./components/show_post.js');
 import {Filters} from './constants/visibilityFilters';
-import {fetchAllPosts} from './actions/posts';
 import { Provider } from 'react-redux';
 import configureStore from './store/store';
 import { connect } from 'react-redux';
-import {createPosts, setSelectedPost, setSelectedPostByName, setSelectedPostById , requestPost, fetchMyPosts} from './actions/posts';
+import {createPosts, setSelectedPost, setSelectedPostByName, setSelectedPostById , requestPost, fetchProfile, fetchPosts} from './actions/posts';
 import * as VisibilityConstants from './constants/visibilityFilters'
-import {setUserData} from './actions/user'
+import {addUserData, setCurrentUserId, getUserData, setProfileId} from './actions/user'
 import {setVisibilityFilter} from './actions/visibilityFilters'
 
 
-//Elements generated with wrapper
-// function getInitialData(){
-// 	var initialState = window.fodoo_data
-// 	return {
-// 		user: initialState.user,
-// 		posts: initialState.posts
-// 	}
-// }
 const store = configureStore();
 
 store.dispatch(createPosts(window.fodoo_data.posts));
-store.dispatch(setUserData(window.fodoo_data.user));
+store.dispatch(setCurrentUserId(window.fodoo_data.user && window.fodoo_data.user.id));
+store.dispatch(addUserData(window.fodoo_data.user));
+store.dispatch(addUserData(window.fodoo_data.profile_user));
+store.dispatch(setProfileId(window.fodoo_data.profile_id));
 store.dispatch(setSelectedPost(window.fodoo_data.selected_post || {}));
 store.dispatch(setVisibilityFilter(window.fodoo_data.posts_visibility));
 
-function selectPosts(posts, filter, my_id, state){
+function selectPosts(posts, filter, state){
 	switch(filter){
 		case VisibilityConstants.Filters.SHOW_ALL:
 			return posts.filter(post => post.active === true);
 		case VisibilityConstants.Filters.SHOW_ALL_APPROVED:
 			return posts.filter(post => (post.approved === true && post.published === true && post.active === true));
-		case VisibilityConstants.Filters.SHOW_MY:
-			if(state.user.admin)
-				return posts.filter(post => post.active === true);;
-			return posts.filter(post => (post.user_id == my_id && post.active === true));
+		case VisibilityConstants.Filters.SHOW_PROFILE:
+			var profile_user = getUserData(state.user.users, state.user.profileId);
```

Comment:
state.user.profileUserId ? 
And how is it ensured that the array state.user.users always has user with id state.user.profileId?